Userland/Applications/PackageManager/main.cpp

```diff
@@ -0,0 +1,186 @@
+        current_view = current_view.substring_view(0, newline_separator_index.value());
+        if (current_view.starts_with("dependency"sv)) {
+            auto parts = current_view.split_view(' ');
+            auto type = Port::Type::Dependency;
+            // FIXME: Add versioning when printing these ports!
+            ports.append(Port { TRY(String::from_utf8(parts[1])), type, TRY(String::from_utf8(""sv)) });
+        } else {
+            auto parts = current_view.split_view(' ');
+            VERIFY(parts.size() == 3);
+            auto type = Port::Type::Auto;
+            if (parts[0] == "manual"sv) {
+                type = Port::Type::Manual;
+            } else if (parts[0] == "auto"sv) {
+                type = Port::Type::Auto;
+            } else {
+                VERIFY_NOT_REACHED();
+            }
+            ports.append(Port { TRY(String::from_utf8(parts[1])), type, TRY(String::from_utf8(parts[2])) });
+        }
+        pointer += current_view.length() + 1;
+    };
+
+    if (!filter_type.has_value()) {
+        for (auto& port : ports) {
+            callback(port);
+        }
+    } else {
+        for (auto& port : ports) {
+            if (filter_type.value() == port.type())
+                callback(port);
+        }
+    }
+
+    return {};
+}
+
+static ErrorOr<void> print_manually_installed_ports()
+{
+    outln("Manually-installed ports:");
+    TRY(print_installed_ports(Port::Type::Manual, [](auto& port) {
+        outln("{}, installed as {}, version {}", port.name(), port.type_as_string_view(), port.version());
+    }));
+    return {};
+}
+
+static ErrorOr<void> print_dependencies_installed_ports()
+{
+    outln("Dependencies-installed ports:");
+    TRY(print_installed_ports(Port::Type::Dependency, [](auto& port) {
+        outln("{}, installed as {}, version {}", port.name(), port.type_as_string_view(), port.version());
+    }));
+    return {};
+}
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio recvfd thread unix rpath cpath wpath"));
+
+    TRY(Core::System::unveil("/tmp/session/%sid/portal/request", "rw"));
+    TRY(Core::System::unveil("/usr/Ports/installed.db"sv, "rwc"sv));
+    TRY(Core::System::unveil("/usr/Ports/packages.db"sv, "rwc"sv));
+    TRY(Core::System::unveil("/res"sv, "r"sv));
+    TRY(Core::System::unveil("/usr/lib"sv, "r"sv));
+    TRY(Core::System::unveil(nullptr, nullptr));
+
+    bool show_all_installed_ports = false;
+    bool show_all_dependency_ports = false;
+    bool update_packages_db = false;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(show_all_installed_ports, "Show all manually-installed ports", "show-all-manual-ports", 'l');
+    args_parser.add_option(show_all_dependency_ports, "Show all dependencies' ports", "show-all-dependency-ports", 'd');
+    args_parser.add_option(update_packages_db, "Sync/Update ports database", "sync-all-ports-database", 'S');
```

Comment:
Can the long versions be a little less wordy and the short version of "update" be "u" instead of pacman's unintuitive uppercase "S"?

Reply:
Sure, I will change the “S” to “u”, although muscle memory is kicking in here to be honest…

Reply:
why not to go for `list`, `deps` and `update` (or be it `sync`, could be an alias too) then…? 😅